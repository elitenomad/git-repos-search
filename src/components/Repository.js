import React from "react"
import Badge from "react-bootstrap/Badge"
import Card from "react-bootstrap/Card"

const Repository = (props) => {
  const {
    repo: {id, name, description, watchers_count},
  } = props

  return (
    <Card>
      <Card.Header as="h2">{name}</Card.Header>
      <Card.Body>
        <Card.Title>{id}</Card.Title>
        <Card.Text>{description}</Card.Text>
        <div className="text-right">
          <Badge variant="info">Watchers count: {watchers_count}</Badge>{" "}
        </div>
      </Card.Body>
    </Card>
  )
}

export default Repository
