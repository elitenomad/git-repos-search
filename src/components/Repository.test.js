import React from "react"
import renderer from "react-test-renderer"
import Repository from "./Repository"

describe("Repository", () => {
  const props = {
    repo: {
      id: "test",
      name: "test",
      description: "testing desc",
      watchers_count: 25,
    },
  }

  test("for given props matches snapshot", () => {
    const tree = renderer.create(<Repository {...props} />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
