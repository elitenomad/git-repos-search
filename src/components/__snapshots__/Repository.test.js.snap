// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repository for given props matches snapshot 1`] = `
<div
  className="card"
>
  <h2
    className="card-header"
  >
    test
  </h2>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      test
    </div>
    <p
      className="card-text"
    >
      testing desc
    </p>
    <div
      className="text-right"
    >
      <span
        className="badge badge-info"
      >
        Watchers count: 
        25
      </span>
       
    </div>
  </div>
</div>
`;
